gpt_engineer/applications/cli/main.py

```diff
@@ -76,8 +76,12 @@ def load_env_if_needed():
         load_dotenv()
     if os.getenv("OPENAI_API_KEY") is None:
         load_dotenv(dotenv_path=os.path.join(os.getcwd(), ".env"))
+
     openai.api_key = os.getenv("OPENAI_API_KEY")
 
+    if openai.api_key == "sk-xxx":
```

Comment:
this doesnt look clean to me. A boolean variable called isLocalModel or LocalModel or something would be better

Reply:
I was thinking about this quite a lot and then decided that the above is cleanest. And requires the least amount of code changes in the code base. At the price of code readability  🤷‍♂️

If we introduce `LocalModel` variable or sth similar then we need to have a list of what clarifies as a local model. There are many options for open LLMs. Model list isn't as trivial as for the OpenAI case.

In principle we could require the user to set another env variable `local_model` and then decide on that.

Not sure. I'm open to suggestions here.

Reply:
@TheoMcCabe, do you have any follow-up comments on @zigabrencic's explanation?